go/libraries/events/event_flush.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2019 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package events
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	eventsapi "github.com/liquidata-inc/dolt/go/gen/proto/dolt/services/eventsapi_v1alpha1"
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/env"
+	"github.com/liquidata-inc/dolt/go/libraries/utils/filesys"
+)
+
+// EventGrpcFlush parses dolt event logs sends the events to the events server
+type EventGrpcFlush struct {
+	em  *GrpcEmitter
+	fbp *FileBackedProc
+}
+
+// ErrEventsDataDir occurs when events are trying to be  flushed, but the events data directory
+// does not yet exist
+var ErrEventsDataDir = errors.New("unable to flush, events data directory does not exist")
+
+func getGRPCEmitter(dEnv *env.DoltEnv) *GrpcEmitter {
+	// actual grpc server config
+
+	// host := dEnv.Config.GetStringOrDefault(env.MetricsHost, env.DefaultMetricsHost)
+	// portStr := dEnv.Config.GetStringOrDefault(env.MetricsPort, env.DefaultMetricsPort)
+	// insecureStr := dEnv.Config.GetStringOrDefault(env.MetricsInsecure, "false")
+
+	// port, err := strconv.ParseUint(*portStr, 10, 16)
+
+	// if err != nil {
+	// 	log.Println(color.YellowString("The config value of '%s' is '%s' which is not a valid port.", env.MetricsPort, *portStr))
+	// 	return emitter
+	// }
+
+	// insecure, err := strconv.ParseBool(*insecureStr)
+
+	// if err != nil {
+	// 	log.Println(color.YellowString("The config value of '%s' is '%s' which is not a valid true/false value", env.MetricsInsecure, *insecureStr))
+	// }
+
+	// hostAndPort := fmt.Sprintf("%s:%d", *host, port)
+	// conn, _ := dEnv.GrpcConnWithCreds(hostAndPort, insecure, nil)
+
+	// local grpc for dev
+	hostAndPort := fmt.Sprintf("%s:%d", "localhost", 50051)
```

Comment:
you can connect locally without having to hard code this.  You can run these commands:

    dolt config --global --add metrics.host localhost
    dolt config --global --add metrics.port 50051
    dolt config --global --add metrics.insecure true

in order to point your dolt cli at your local metrics server.